From c862ba33f92e1d814a7dd0856f87d53625911181 Mon Sep 17 00:00:00 2001
From: "Frederic Pierret (Epitre)" <frederic.epitre@orange.fr>
Date: Tue, 19 Sep 2017 20:24:23 +0200
Subject: [PATCH 23/31] anaconda: don't crash when no target disk is available

self.storage.root_device may be None in such case. Instead, allow proper reporting that no space is available.
---
 pyanaconda/ui/lib/space.py | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/pyanaconda/ui/lib/space.py b/pyanaconda/ui/lib/space.py
index abc4cce..421578d 100644
--- a/pyanaconda/ui/lib/space.py
+++ b/pyanaconda/ui/lib/space.py
@@ -75,8 +75,11 @@ class FileSystemSpaceChecker(object):
         log.info("fs space: %s  needed: %s", free, needed)
         self.success = (free > needed)
         if not self.success:
-            dev_required_size = self.payload.requiredDeviceSize(self.storage.root_device.format)
-            self.deficit = dev_required_size - self.storage.root_device.size
+            if self.storage.root_device:
+                dev_required_size = self.payload.requiredDeviceSize(self.storage.root_device.format)
+                self.deficit = dev_required_size - self.storage.root_device.size
+            else:
+                self.deficit = needed - free
             self.error_message = _(self.error_template) % self.deficit
 
         return self.success
@@ -107,8 +110,11 @@ class DirInstallSpaceChecker(FileSystemSpaceChecker):
         log.info("fs space: %s  needed: %s", free, needed)
         self.success = (free > needed)
         if not self.success:
-            dev_required_size = self.payload.requiredDeviceSize(self.storage.root_device.format)
-            self.deficit = dev_required_size - self.storage.root_device.size
+            if self.storage.root_device:
+                dev_required_size = self.payload.requiredDeviceSize(self.storage.root_device.format)
+                self.deficit = dev_required_size - self.storage.root_device.size
+            else:
+                self.deficit = needed - free
             self.error_message = _(self.error_template) % self.deficit
 
         return self.success
-- 
2.9.5

